marshmallow/fields.py

```diff
@@ -1016,32 +1027,86 @@ def _deserialize(self, value, attr, data):
 class Date(Field):
     """ISO8601-formatted date string.
 
+    :param strict: If True, only accept Date objets. If False, will accept any
```

Comment:
Drive-by: There is some discussion about removing `strict` flags on fields here: https://github.com/marshmallow-code/marshmallow/pull/755#issuecomment-377662237 , so I have reservations about adding it to `Date`.